lib/Twig/Environment.php

```diff
@@ -112,6 +117,23 @@ public function __construct(Twig_LoaderInterface $loader = null, $options = arra
         $this->addExtension(new Twig_Extension_Escaper($options['autoescape']));
         $this->addExtension(new Twig_Extension_Optimizer($options['optimizations']));
         $this->staging = new Twig_Extension_Staging();
+
+        // For BC
+        if (is_string($this->originalCache)) {
+            $r = new ReflectionMethod($this, 'writeCacheFile');
+            if ($r->getDeclaringClass() === get_class($this)) {
```

Comment:
you should rather check ``!== __CLASS__`` here, to account for different level of inheritance